### ISSUE TITLE
Port `twisted.internet.task` to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6052]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#6052|
|Type|enhancement|
|Created|2012-10-04 13:27:12Z|
|Branch|https://github.com/twisted/twisted/tree/task-py3-6052|

`twisted.internet.task` should run on Python 3.

<details><summary>Searchable metadata</summary>

```
trac-id__6052 6052
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_task_py3_6052 branches/task-py3-6052
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1349357232000000 1349357232000000
changetime__1349375447000000 1349375447000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst set owner to @exarkun|
|-|-|

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/task-py3-6052

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst commented|
|-|-|

(In [35936]) Branching to 'task-py3-6052'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @itamarst|
|-|-|

Looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set status to `closed`|
|-|-|

(In [35951]) Merge task-py3-6052.

Author: itamar
Review: exarkun
Fixes: [#6052](https://github.com/twisted/twisted/issues/6052)

twisted.internet.task now runs on Python 3.
